DataFormats/TrackReco/interface/TrackBase.h

```diff
@@ -596,16 +602,29 @@ namespace reco {
   inline double TrackBase::d0() const { return -dxy(); }
 
   // dsz parameter (THIS IS NOT the SZ impact parameter to (0,0,0) if refPoint is far from (0,0,0): see parametrization definition above for details)
-  inline double TrackBase::dsz() const { return vz() * pt() / p() - (vx() * px() + vy() * py()) / pt() * pz() / p(); }
+  inline double TrackBase::dsz() const {
+    const auto thepinv = 1/p();
+    const auto theptoverp = pt() * thepinv;
```

Comment:
`     const auto theptoverp = sqrt ( pt2() / p2() );`
apparently saves one sqrt.
Also, shouldn't 1/pt*p be also faster with `1/sqrt(pt2*p2)`?

Reply:
can this part of the review be addressed?

Reply:
note that this will also make the updated code more self-consistent with changes made in another method `TrackBase::dsz(const Point &myBeamSpot)`